packages/xrpl/src/models/methods/nftHistory.ts

```diff
@@ -0,0 +1,117 @@
+import { LedgerIndex, ResponseOnlyTxInfo } from '../common'
+import { Transaction, TransactionMetadata } from '../transactions'
+
+import { BaseRequest, BaseResponse } from './baseMethod'
+
+/**
+ * The nft_history method retrieves a list of transactions that involved the
+ * specified NFToken. Expects a response in the form of a {@link
+ * NFTHistoryResponse}.
+ *
+ * @category Requests
+ */
+export interface NFTHistoryRequest extends BaseRequest {
+  command: 'nft_history'
+  /**
+   * The unique identifier of an NFToken.
+   */
+  nft_id: string
+  /**
+   * Use to specify the earliest ledger to include transactions from. A value
+   * of -1 instructs the server to use the earliest validated ledger version
+   * available.
+   */
+  ledger_index_min?: number
+  /**
+   * Use to specify the most recent ledger to include transactions from. A
+   * value of -1 instructs the server to use the most recent validated ledger
+   * version available.
+   */
+  ledger_index_max?: number
+  /** Use to look for transactions from a single ledger only. */
+  ledger_hash?: string
+  /** Use to look for transactions from a single ledger only. */
+  ledger_index?: LedgerIndex
+  /**
+   * If true, return transactions as hex strings instead of JSON. The default is
+   * false.
+   */
+  binary?: boolean
+  /**
+   * If true, returns values indexed with the oldest ledger first. Otherwise,
+   * the results are indexed with the newest ledger first.
+   */
+  forward?: boolean
+  /**
+   * Default varies. Limit the number of transactions to retrieve. The server
+   * is not required to honor this value.
+   */
+  limit?: number
+  /**
+   * Value from a previous paginated response. Resume retrieving data where
+   * that response left off. This value is stable even if there is a change in
+   * the server's range of available ledgers.
+   */
+  marker?: unknown
+}
+
+interface NFTokenTransaction {
+  /** The ledger index of the ledger version that included this transaction. */
+  ledger_index: number
+  /**
+   * If binary is True, then this is a hex string of the transaction metadata.
+   * Otherwise, the transaction metadata is included in JSON format.
+   */
+  meta: string | TransactionMetadata
+  /** JSON object defining the transaction. */
+  tx?: Transaction & ResponseOnlyTxInfo
+  /** Unique hashed String representing the transaction. */
+  tx_blob?: string
+  /**
+   * Whether or not the transaction is included in a validated ledger. Any
+   * transaction not yet in a validated ledger is subject to change.
+   */
+  validated: boolean
+}
```

Comment:
is there not an object that represents a generic tx you can already use? what is being used for account_tx?